### DISCUSSION TITLE
prisma-plugin: What is the correct way for defining and using enums?

### DISCUSSION THREAD

**Discussion by lockinmarv** [hayes/pothos#316] [Q&A] [answered]
Hey!

I have an enum `Role` defined in my `schema.prisma` like this
```prisma
enum Role {
  USER
  ADMIN
}
```
but I'm not 100% sure what the correct implementation would be. Is this correct?
```ts
builder.enumType(Role, {
  name: 'Role'
})

builder.prismaObject('User', {
  findUnique: (user) => ({ id: user.id }),
  fields: (t) => ({
    // ... other fields
    role: t.field({
      type: Role,
      resolve: ({ role }) => role
    })
  })
})
```

**Comment by hayes** [comment, marked as answer]
This looks correct, is this causing an issue for you?

**Reply by lockinmarv** [reply]
No not at all, I was just wondering if there is an auto-generated type for it. 

But that's fine, I've probably misunderstood something. Thanks!

**Reply by hayes** [reply]
Pothos tends to avoid auto-generating anywhere whenever possible.  The goal is to make it easy to define things yourself, so that you have full control, and can configure things to work the way you want

**Reply by lockinmarv** [reply]
Gotcha, thanks for the clarification!
